core/includes/form.inc

```diff
@@ -3333,12 +3342,19 @@ function form_process_checkboxes($element) {
       // sub-elements.
       $weight += 0.001;
 
+      // Only enabled checkboxes receive their values from the form submission;
+      // disabled checkboxes use their default value.
+      $default_value = NULL;
+      if (isset($value[$key]) || (in_array($key, $element['#default_value']) && isset($element[$key]['#disabled']))) {
```

Comment:
The issue I described in https://github.com/backdrop/backdrop-issues/issues/4083#issuecomment-535437002 is caused by the `in_array()` condition. For some reason I don't understand (but which is detailed here: https://www.php.net/manual/en/function.in-array.php#106319)) we need to do strict checking of the array, not loose checking. This is done by setting the `in_array()`'s third parameter to `TRUE`:

```suggestion
      if (isset($value[$key]) || (in_array($key, $element['#default_value'], TRUE) && isset($element[$key]['#disabled']))) {
```